libraries/AP_Compass/AP_Compass.h

```diff
@@ -166,22 +177,22 @@ friend class AP_Compass_Backend;
     bool consistent() const;
 
     /// Return the health of a compass
-    bool healthy(uint8_t i) const { return _state[i].healthy; }
-    bool healthy(void) const { return healthy(get_primary()); }
+    bool healthy(uint8_t i) const { return _get_state(Priority(i)).healthy; }
+    bool healthy(void) const { return healthy(0); }
     uint8_t get_healthy_mask() const;
 
     /// Returns the current offset values
     ///
     /// @returns                    The current compass offsets in milligauss.
     ///
-    const Vector3f &get_offsets(uint8_t i) const { return _state[i].offset; }
-    const Vector3f &get_offsets(void) const { return get_offsets(get_primary()); }
+    const Vector3f &get_offsets(uint8_t i) const { return _get_state(Priority(i)).offset; }
+    const Vector3f &get_offsets(void) const { return get_offsets(0); }
```

Comment:
Sorry if I'm misunderstanding but previously I think get_offsets() (with no argument field) would return the highest priority compass.  I think now though it returns the 0th compass.  Maybe I'm just confused about what "Priority()" does.  Does it take an instance and convert it to a priority?  or does it take a Priority and convert it to an instance?  Maybe as mentioned in another comment we just need to renamed "Priority" to make it more clear what it's converting to what.

Reply:
To Consumers of Compass frontend now 0 == Primary. There's no distinction between primary and 0th instance and secondary and 1st instance and so forth. We take care of that in the frontend.